int64 mode
int64 flip
---
int64 acknowledge